quora-api/src/main/java/com/upgrad/quora/api/controller/QuestionController.java

```diff
@@ -0,0 +1,103 @@
+import com.upgrad.quora.service.exception.AuthenticationFailedException;
+import com.upgrad.quora.service.exception.AuthorizationFailedException;
+import com.upgrad.quora.service.exception.UserNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/")
+public class QuestionController {
+
+    @Autowired
+    private QuestionsService questionsService;
+    @Autowired
+    private AuthenticationService authenticationService;
+
+    @Autowired
+    private UserDao userDao;
+
+    /** Implementation of GetAllQuestions endpoint.
+     * This method expose end point /question/all
+     *
+     * @param authorization : authorization token.
+     * @return List<QuestionDetailsResponse>
+     * @throws AuthorizationFailedException
+     */
+    @RequestMapping(method = RequestMethod.GET, path = "/question/all", produces = MediaType.APPLICATION_JSON_UTF8_VALUE )
+    public ResponseEntity<List<QuestionDetailsResponse>> getAllQuestions(@RequestHeader("authorization") final String accessToken) throws AuthorizationFailedException {
+
+        UserAuthTokenEntity authTokenEntity =  authenticationService.getUserByToken(accessToken);
+        if (authTokenEntity != null) {
+
+            List<QuestionEntity> allQuestions = questionsService.getAllQuestions(accessToken);
+
+            return getListResponseEntity(allQuestions);
+        }
+        else
+        {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+    }
+
+    /** Handled endpoint to display all questions. Prepares the List of All question to be displayed.
+     * This method expose end point /question/all
+     *
+     * @param allQuestion : List of QuestionEntity
+     * @return ResponseEntity
+     *
+     */
+    private ResponseEntity<List<QuestionDetailsResponse>> getListResponseEntity(List<QuestionEntity> allQuestions) {
+        QuestionEntity questionEntity;
+        if (allQuestions.size() != 0) {
+            List<QuestionDetailsResponse> displayAllQuestions = new ArrayList<>();
+            for (int i = 0; i < allQuestions.size(); i++) {
+                questionEntity = allQuestions.get(i);
+                QuestionDetailsResponse questionDetailsResponse = new QuestionDetailsResponse().id(questionEntity.getUuid()).content(questionEntity.getContent());
+                displayAllQuestions.add(questionDetailsResponse);
+            }
+            return new ResponseEntity<List<QuestionDetailsResponse>>(displayAllQuestions, HttpStatus.OK);
+        }
+        else {
+            return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+        }
+    }
+
+    /** Implementation of GetAllQuestions endpoint.
+     * This method expose end point /question/all
```

Comment:
Comment needs update to reflect get user by ID and not ALL

Reply:
Comments. Updated